src/entity.ts

```diff
@@ -857,6 +857,7 @@ export namespace entity {
       if (
         firstPathPartIsArray &&
         // check also if the property in question is actually an array value.
+        entity.properties![firstPathPart] &&
```

Comment:
I see no harm in this. If we don't check for this and `entity.properties![firstPathPart]` ever happens to be undefined then this will throw an error and I can't imagine that being desirable.